simple_salesforce/tests/test_api.py

```diff
@@ -604,7 +604,111 @@ def test_query(self):
                             session=session)
 
         result = client.query('SELECT ID FROM Account')
-        self.assertEqual(result, {})
+        self.assertEqual(result, {})\
+
+    @responses.activate
+    def test_query_parse_float(self):
+        """Test parsing of floating point numbers as floats"""
+        responses.add(
+            responses.GET,
+            re.compile(r'^https://.*/query/\?q=SELECT\+ID%2C\+Price\+'
+                       r'FROM\+Account$'),
+            body='{"records": [{"ID": "1", "Price": 13.40},'
+                              '{"ID": "2", "Price": 1.12345678901234567},'
+                              '{"ID": "3", "Price": 123456789012345678},'
+                              '{"ID": "4", "Price": 0},'
+                              '{"ID": "5", "Price": -1.1234567890123456},'
+                              '{"ID": "6", "Price": -12345678901234567}],'
+                 '"done": false, "nextRecordsUrl": '
+                 '"https://example.com/query/next-records-id", "totalSize": 1}',
+            status=http.OK)
+        session = requests.Session()
+        client = Salesforce(session_id=tests.SESSION_ID,
+                            instance_url=tests.SERVER_URL,
+                            session=session)
+
+        result = client.query('SELECT ID, Price FROM Account')
+        print(result)
+        self.assertEqual(
+            result,
+            OrderedDict([
+                ('records', [
+                    OrderedDict([('ID', '1'), ('Price', 13.4)]),
+                    OrderedDict([
+                        ('ID', '2'), ('Price', 1.12345678901234567)
+                    ]),
+                    OrderedDict([
+                        ('ID', '3'), ('Price', 123456789012345678)
+                    ]),
+                    OrderedDict([('ID', '4'), ('Price', 0)]),
+                    OrderedDict([
+                        ('ID', '5'), ('Price', -1.1234567890123456),
+                    ]),
+                    OrderedDict([
+                        ('ID', '6'), ('Price', -12345678901234567),
+                    ]),
+                ]),
+                ('done', False),
+                ('nextRecordsUrl', "https://example.com/query/next-records-id"),
+                ('totalSize', 1),
```

Comment:
While not used by the test, `totalSize` should match the actual number of records (six)